From f0f5c415a6e0abc40514f97113deb52a343174ee Mon Sep 17 00:00:00 2001
From: Pascal de Bruijn <pmjdebruijn@pcode.nl>
Date: Thu, 30 Jan 2020 13:48:44 +0100
Subject: [PATCH] fix cross compilation failure due to size_t typecast in
 define
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The following commit introduces a cross-compilation failure:

   93c728b77cfb47f5cfdd1863f8982ea59d344205
  "Try to control ZEND_MM_ALIGNED_SIZE type"

br-arm-full/build/php-7.4.2/Zend/zend_alloc.h:30:38:
error: missing binary operator before token "8"
                                              ^
br-arm-full/build/php-7.4.2/ext/opcache/ZendAccelerator.c:1380:7:
note: in expansion of macro ‘ZEND_MM_ALIGNMENT’

Closes GH-5128.

Signed-off-by: Pascal de Bruijn <p.debruijn@unilogic.nl>
---
 Zend/zend_alloc.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Zend/zend_alloc.h b/Zend/zend_alloc.h
index c5838f2ac684..8bb854328dbd 100644
--- a/Zend/zend_alloc.h
+++ b/Zend/zend_alloc.h
@@ -27,12 +27,12 @@
 #include "zend.h"
 
 #ifndef ZEND_MM_ALIGNMENT
-# define ZEND_MM_ALIGNMENT ((size_t) 8)
+# define ZEND_MM_ALIGNMENT Z_UL(8)
 # define ZEND_MM_ALIGNMENT_LOG2 Z_L(3)
 #elif ZEND_MM_ALIGNMENT < 4
 # undef ZEND_MM_ALIGNMENT
 # undef ZEND_MM_ALIGNMENT_LOG2
-# define ZEND_MM_ALIGNMENT ((size_t) 4)
+# define ZEND_MM_ALIGNMENT Z_UL(4)
 # define ZEND_MM_ALIGNMENT_LOG2 Z_L(2)
 #endif
 
